File: 2_add_two_numbers/2_add_two_numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        result = ListNode(0)
        node = result
        extra = 0
        while l1 or l2 or extra:
            if l1 == None:
                l1 = ListNode(0)
            if l2 == None:
                l2 = ListNode(0)
                
            node.next = ListNode((l1.val + l2.val + extra) % 10)
            extra = (l1.val + l2.val + extra) // 10
            node = node.next
            l1 = l1.next
            l2 = l2.next
        
        return result.next
                
            

File: 2_add_two_numbers/2_add_two_numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
	listNode ans(0);
	listNode *node = &ans;
	int extra = 0;
	
	while (l1 || l2 || extra) {
		int l1Value = 0;
		int l2Value = 0;
		if (l1 ) {
			l1Value = l1->val;
			l1 = l1->next;
		}
		if (l2 ) {
			l2Value = l2->val;
			l2 = l2->next;
		}
		nede->next = new ListNode((l1Value + l2Value + extra) % 10);
		extra = (l1Value + l2Value + extra) / 10;
		node = node -> next;
	}
	return ans.next;
    }
};
